### PR TITLE
Breez Liquid SDK API Key integration

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
This PR improves API key handling, ensure secure storage across platforms and streamline the reconnection process.

### Changelog:
- **Android:** Get `API_KEY` from compile-time variables on Android f849bd7
- Set SDK Config's `breezApiKey` to `API_KEY` from environment variables 5240097
- Throw an exception if `API_KEY` is not set in environment variables 83d304a
- Add readme on how to define environment variables b6a4595
- Remove redundant secure storage access when reconnecting on launch  5f1c7b0
- Store `breezApiKey` to secure storage b44a052
- **iOS:** Access `API_KEY` from shared secure storage on iOS notification service app extension a511e42

### CODE REVIEW

File: packages/sdk_connectivity_cubit/lib/src/sdk_connectivity_cubit.dart [hunk 1]
```diff
@@ -39,7 +39,7 @@ class SdkConnectivityCubit extends Cubit<SdkConnectivityState> {
     }
   }
 
-  Future<void> _connect(String mnemonic, {bool storeMnemonic = false}) async {
+  Future<void> _connect(String mnemonic, {bool storeCredentials = false}) async {
```
Comment (roeierez) [inline packages/sdk_connectivity_cubit/lib/src/sdk_connectivity_cubit.dart:42]: It seems that on upgrade we won't save the API key(only on restore and register). Why not saving this on each connect?

**Review by roeierez — CHANGES_REQUESTED** [review]
Looks good, dropped one comment.

File: packages/sdk_connectivity_cubit/lib/src/sdk_connectivity_cubit.dart (hunk 1 above)
Comment (erdemyerebasmaz) [reply packages/sdk_connectivity_cubit/lib/src/sdk_connectivity_cubit.dart:42]: There was an IDE bug and I forgot to revert these changes after refactoring.

API key will be stored on each connect if it has changed or does not exist in secure storage.

Comment (erdemyerebasmaz) [reply packages/sdk_connectivity_cubit/lib/src/sdk_connectivity_cubit.dart]: Done. 
[Store breezApiKey on every connect](https://github.com/breez/misty-breez/pull/196/commits/4feedfc1b115247d5ec15b67aa9a6e7db795515b)

Comment (erdemyerebasmaz) [reply packages/sdk_connectivity_cubit/lib/src/sdk_connectivity_cubit.dart]: and also [simplified _storeBreezApiKey logic](https://github.com/breez/misty-breez/pull/196/commits/7635876d7f0d283716be9f26783ec7b08219962c)

**Review by roeierez — APPROVED** [review]

**Review by dangeross — APPROVED** [review]
LGTM! Don't forget the API_KEY in the github secrets

**Pull request merged**